shared/__init__.py

```diff
@@ -1,7 +1,7 @@
 from dataclasses import dataclass, field
 from datetime import datetime
 from functools import partial
-from typing import Any, Dict, List, Optional, Set, Tuple
+from typing import Any, Dict, Optional, List, Set, Tuple
```

Comment:
FWIW, this was a result of running the linting. I agree it's unrelated but it's possibly something desirable